homeassistant/components/fitbit/sensor.py

```diff
@@ -627,86 +680,46 @@ def __init__(
     @property
     def icon(self) -> str | None:
         """Icon to use in the frontend, if any."""
-        if (
-            self.entity_description.key == "devices/battery"
-            and self.extra is not None
-            and (extra_battery := self.extra.get("battery")) is not None
-            and (battery_level := BATTERY_LEVELS.get(extra_battery)) is not None
-        ):
-            return icon_for_battery_level(battery_level=battery_level)
+        if self.entity_description.key == "devices/battery" and self.device is not None:
+            return icon_for_battery_level(battery_level=self.device.battery_level)
         return self.entity_description.icon
 
     @property
     def extra_state_attributes(self) -> dict[str, str | None]:
         """Return the state attributes."""
         attrs: dict[str, str | None] = {}
 
-        if self.extra is not None:
-            attrs["model"] = self.extra.get("deviceVersion")
-            extra_type = self.extra.get("type")
-            attrs["type"] = extra_type.lower() if extra_type is not None else None
+        if self.device is not None:
+            attrs["model"] = self.device.device_version
+            device_type = self.device.type
+            attrs["type"] = device_type.lower() if device_type is not None else None
 
         return attrs
 
     def update(self) -> None:
         """Get the latest data from the Fitbit API and update the states."""
         resource_type = self.entity_description.key
-        if resource_type == "devices/battery" and self.extra is not None:
-            registered_devs: list[dict[str, Any]] = self.client.get_devices()
-            device_id = self.extra.get("id")
-            self.extra = list(
-                filter(lambda device: device.get("id") == device_id, registered_devs)
+        if resource_type == "devices/battery" and self.device is not None:
+            device_id = self.device.id
+            registered_devs: list[FitbitDevice] = self.api.get_devices()
+            self.device = list(
+                filter(
+                    lambda device: device.id == device_id,
```

Comment:
Please use a list comprehension instead of filter + lambda.

Reply:
Done.